packages/fileeditor-extension/src/index.ts

```diff
@@ -735,24 +735,24 @@ function activate(
     menu.runMenu.codeRunners.add({
       tracker,
       noun: 'Code',
-      isEnabled: current => {
-        let found = false;
-        consoleTracker.forEach(console => {
-          if (console.console.session.path === current.context.path) {
-            found = true;
-          }
-        });
-        return found;
-      },
+      isEnabled: current =>
+        !!consoleTracker.find(
+          c => c.console.session.path === current.context.path
+        ),
       run: () => commands.execute(CommandIDs.runCode),
       runAll: () => commands.execute(CommandIDs.runAllCode),
       restartAndRunAll: current => {
-        return current.context.session.restart().then(restarted => {
-          if (restarted) {
-            void commands.execute(CommandIDs.runAllCode);
-          }
-          return restarted;
-        });
+        const console = consoleTracker.find(
+          console => console.console.session.path === current.context.path
+        );
+        if (console) {
+          return console.session.restart().then(restarted => {
```

Comment:
Here you use `console.session`, but above you use `console.console.session`. Is there a reason for using one or the other? Can we just use one for consistency?

Reply:
in the isEnabled we use `console.session`, so perhaps that has two votes here...

Reply:
(oh, I guess it actually uses `c.console.session`, which is equivalent to `console.console.session`...)

Reply:
No, no reason. I think at some point we discussed removing the one from the `widget` because it was confusing, but never wound up doing that.